docs/specs/NETWORKING.md

```diff
@@ -1,3 +1,769 @@
+It holds relevant registry data and the validators consensus data.
+
+`Bootnode` is a public peer which is responsible for helping new peers to find other peers in the network.
+It has a stable ENR that is provided with default configuration, so other peers could join the network easily.
+
+`Exporter` is a public peer that is responsible for collecting and exporting information from the network. \
+It collects registry data and consensus data (decided messages) of all the validators in the network. \
+It has a stable ENR that is provided with default configuration, 
+so it will have a stable connection with all peers and won't be affected by scoring, pruning, backoff etc.
+
+
+### Identity
+
+Identity in the network is based on two types of keys:
+
+`Network Key` is used to create network/[libp2p identity](https://docs.libp2p.io/concepts/peer-id) (`peer.ID`), 
+will be used by all network peers to setup a secured connection. \
+Unless provided, the key will be generated and stored locally for future use, 
+and can be revoked in case it was compromised. 
+
+`Operator Key` is used for decryption of share's keys that are used for signing/verifying consensus messages and duties. \
+Exporter and Bootnode does not hold this key.
+
+In addition, Operator nodes will expose an `Operator ID` that is calculated from the operator public key, 
+and helps to identify a peer in the network.
+
+### Network Discovery
+
+[discv5](https://github.com/ethereum/devp2p/blob/master/discv5/discv5.md) 
+is used in `SSV.Network` as the discovery system.
+
+More information is available in [Networking > Discovery](#discovery)
+
+
+### Peer Scoring
+
+The peer scoring in `SSV.Network` consists of the following types of scorers:
+
+- [Connection Scoring](#connection-scoring) - calculated during handshake, affects the peers we connect to
+- [Message Scoring](#message-scoring) - tracks operators behavior w.r.t incoming IBFT messages
+- [Pubsub Scoring](#pubsub-scoring) - calculated by libp2p, tracks over gossip tasks and pubsub related metrics
+
+For more info please refer to the linked sections.
+
+------
+
+
+## Wire
+
+Network interaction includes several types of protocols:
+
+## Consensus Protocol
+
+`IBFT`/`QBFT` consensus protocol is used to govern `SSV` network.
+`IBFT` ensures that consensus can be reached by a committee of `n` 
+operator nodes while tolerating a certain amount of `f` faulty nodes as defined by `n ≥ 3f + 1`.
+
+As part of the algorithm, nodes are exchanging messages with other nodes in the committee. \
+Once the committee reaches consensus, the nodes will publish the decided message across the network.
+
+Consensus messages are being sent in the network over pubsub topics (see [subnets](#subnets))
+
+More information regarding the protocol can be found in [iBFT annotated paper (By Blox)](/ibft/IBFT.md)
+
+### Message Structure
+
+`SignedMessage` is a wrapper for IBFT messages, it holds a message and its signature with a list of signer IDs:
```

Comment:
I think we should reference the in-dev QBFT and SSV specs
https://github.com/bloxapp/ssv/blob/ssv_spec/docs/spec/qbft/messages.go

Reply:
Also, as this is the network spec we should maybe define a network message struct in which different types of messages are set?

Reply:
We can follow this
https://github.com/bloxapp/ssv/blob/ssv_spec/docs/spec/types/messages.go#L64-L68